### PR TITLE
[mqtt] generic-mqtt-tester tokio1 upgrade

### PR DESCRIPTION
**Pull request by dmolokanov**
Upgrades `generic-mqtt-tester` to tokio1 ecosystem.
Use local dependencies from local path instead of pinning

### CODE REVIEW

File: test/modules/generic-mqtt-tester/Cargo.toml [hunk 1]
```diff
@@ -13,25 +13,25 @@ branch = "span-fix"
...
-mqtt-util = { git = "https://github.com/Azure/iotedge", rev = "4cd508" }
-mqtt-broker-tests-util = { git = "https://github.com/Azure/iotedge", rev = "4cd508" }
-mqtt3 = { git = "https://github.com/Azure/iotedge", rev = "4cd508" }
-trc-client = { path = "../../rust-test-util/test-result-reporting-client" }
\ No newline at end of file
+edgelet-client = { path = "../../../mqtt/edgelet-client" }
+mqtt-util = { path = "../../../mqtt/mqtt-util" }
+mqtt-broker-tests-util = { path = "../../../mqtt/mqtt-broker-tests-util" }
+mqtt3 = { path = "../../../mqtt/mqtt3" }
+trc-client = { path = "../../rust-test-util/test-result-reporting-client" }
```
Comment (nyanzebra) [inline test/modules/generic-mqtt-tester/Cargo.toml:37]: So we don't want to pin to commits?

**Review by and-rewsmith — DISMISSED** [review]

File: test/modules/generic-mqtt-tester/src/main.rs [hunk 2]
```diff
@@ -59,13 +58,14 @@ async fn listen_for_shutdown() -> Result<(), MessageTesterError> {
...
     let mut interrupt =
         signal(SignalKind::interrupt()).map_err(MessageTesterError::CreateUnixSignalListener)?;
-    let mut term =
+    let mut terminate =
         signal(SignalKind::terminate()).map_err(MessageTesterError::CreateUnixSignalListener)?;
-    let interrupt_fut = interrupt.next();
-    let term_fut = term.next();
+    let interrupt = interrupt.recv();
+    let terminate = terminate.recv();
+    pin_mut!(interrupt, terminate);
```
Comment (nyanzebra) [inline test/modules/generic-mqtt-tester/src/main.rs:65]: I am trying to read up on changes in signal, but why do we need to pin the futures now?

File: test/modules/generic-mqtt-tester/src/message_initiator.rs [hunk 3]
```diff
@@ -91,10 +91,21 @@ impl MessageInitiator {
...
+                self.batch_id.to_string(),
+                seq_num,
+            );
+
+            let test_type = trc_client::TestType::Messages;
+            let created_at = chrono::Utc::now();
+            self.reporting_client
+                .report_result(SEND_SOURCE.to_string(), result, test_type, created_at)
+                .await
+                .map_err(MessageTesterError::ReportResult)?;
```
Comment (nyanzebra) [inline test/modules/generic-mqtt-tester/src/message_initiator.rs:105]: Why did we take this out of the helper fn? Did the helper not seem worth it?

Comment (dmolokanov) [reply test/modules/generic-mqtt-tester/src/message_initiator.rs:105]: borrow checker was complaining about the future still holds mutable reference to self. so I just inlined the whole thing

File: test/modules/generic-mqtt-tester/Cargo.toml (hunk 1 above)
Comment (dmolokanov) [reply test/modules/generic-mqtt-tester/Cargo.toml:37]: actually we do not. I did it as a temp fix while upgrading the codebase

File: test/modules/generic-mqtt-tester/src/main.rs (hunk 2 above)
Comment (dmolokanov) [reply test/modules/generic-mqtt-tester/src/main.rs:65]: Signal is not `Stream` anymore. https://docs.rs/tokio/1.5.0/tokio/signal/unix/struct.Signal.html so we need to ensure future returned by `recv` is pinned

Comment (nyanzebra) [reply test/modules/generic-mqtt-tester/src/main.rs:65]: Ah that makes sense

File: test/modules/generic-mqtt-tester/src/message_initiator.rs (hunk 3 above)
Comment (nyanzebra) [reply test/modules/generic-mqtt-tester/src/message_initiator.rs:105]: hmm interesting, but okay works for me

**Review by nyanzebra — DISMISSED** [review]

**Review by vadim-kovalyov — APPROVED** [review]
🚀

**Review by and-rewsmith — APPROVED** [review]
Thanks for upgrading :)

**Pull request merged**